src/shared/components/modal/ModalProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { ModalPortal } from '../../../ModalPortal';
+
+type ModalContextType = {
+  props?: React.ReactElement;
+  setProps?: (children: React.ReactElement) => void;
+};
+
+export const ModalContext = React.createContext<ModalContextType>({});
+export const ModalProvider: React.FC<ModalContextType> = ({ children }) => {
+  const [context, setContext] = React.useState<React.ReactElement>();
```

Comment:
Use better variable name here. state here is not context it's actual Modal component

Reply:
changed it to `modal` and `setModal`